### PR TITLE
feat(plugin): support id as a function (asyncID)

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
The option id as function is compatible but with this commit will also receive context as parameter.

Resolve #32

### CODE REVIEW

File: lib/plugin.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import Vue from 'vue'
 import VueAnalytics from 'vue-analytics'
 
-export default (ctx, inject) => {
+export default async (ctx, inject) => {
   const options = <%= serialize(options) %>
 
+  if (typeof options.id === 'function') {
```
Comment (pi0) [inline lib/plugin.js]: This `id` option overrides vue-analytics `id` option. I suggest at least rename it to something like `asyncId`

Comment (ricardogobbosouza) [reply lib/plugin.js]: What's the problem with overrides?

Comment (pi0) [reply lib/plugin.js]: We are changing library consideration by passing different argument (ctx) at any point vue analytics itself may change signature.

Also async at plugin level is not a good perf idea to advice. Analytics should happen with low priority than real page content.

Comment (ricardogobbosouza) [reply lib/plugin.js]: Ok, changed

**Comment by ricardogobbosouza** [comment]
@pi0 
Any other suggestions?

**Pull request merged**